### PR TITLE
Feature/216 improve docs

### LINKED ISSUE #216: Rewrite readme in form of a specification for apps
- [x] Describe all endpoints HTTP gateway provides
- [x] Describe all headers HTTP gateway processes
- [x] Describe all request parameters and options

### PR DESCRIPTION
**Pull request by KirillovDenis**
close #216

### CODE REVIEW

**Comment by KirillovDenis** [comment]
See how it looks like https://github.com/KirillovDenis/neofs-http-gw/blob/feature/216-improve_docs/docs/spec.md

**Comment by alexvanin** [comment]
Any ideas for improvements? @dansingjulia @acheyda

**Comment by KirillovDenis** [comment]
Maybe we should also mention the [expiration headers](https://github.com/nspcc-dev/neofs-http-gw/blob/master/README.md#uploading) more specifically. But we have https://github.com/nspcc-dev/neofs-http-gw/issues/214 and such headers probably be gone.

File: docs/spec.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+###### Headers
+
+| Header                | Description                                                                                                                                   |
+|-----------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|
+| Common headers        | See [bearer token](#bearer-token).                                                                                                            |
+| `X-Attribute-Neofs-*` | Used to set system NeoFS object attributes <br/> (e.g. use "X-Attribute-Neofs-Expiration-Epoch" to set `__NEOFS__EXPIRATION_EPOCH` attribute) |
+| `X-Attribute-*`       | Used to set regular object attributes <br/> (e.g. use "X-Attribute-My-Tag" to set `My-Tag` attribute)                                         |
+
+If you don't specify the `X-Attribute-Timestamp` header the `Timestamp` attribute can be set anyway
+(see http-gw [configuration](gate-configuration.md)).
```
Comment (masterSplinter01) [inline docs/spec.md:66]: Maybe make the mention more exact? I mean, make a link to `upload-header section`

File: docs/spec.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+### Methods
+
+#### GET
+
+Find objects by prefix for `FilePath` attributes. Return found objects in zip archive.
+Name of files in archive sets to `FilePath` attribute of objects.
+Time of files sets to time when object has started downloading.
+
+Archive can be compressed (see http-gw [configuration](gate-configuration.md)).
```
Comment (masterSplinter01) [inline docs/spec.md:268]: The same

**Comment by masterSplinter01** [comment]
Also, a word `section` in headers: `wallet section`, `peers section`, `logger section` etc in [gate-configuration.md](https://github.com/nspcc-dev/neofs-http-gw/blob/master/docs/gate-configuration.md) looks redundant, doesn't it?

**Comment by acheyda** [comment]
1)If possible, could you please describe the functionality with "?download" in more detail, that it can be used with /get_by_attribute/{cid}/{attr_key}/{attr_val} and /zip/{cid}/{prefix}.

2)In addition, I would like to describe in more detail the functionality of Download zip. For example, by specifying /zip/{cid}/?download, you can download all objects and directories with the FilePath attribute.

3)And it would be great if you mentioned the limitations of NNS in something like this: 
In all download/upload routes you can use container name instead of its cid.
NNS-names can contain only letters (a–z), digits (0-9) no more than 63 characters and a minus sign (-).
NNS-name cannot contain the following characters: (,) (~), (:), (!), (@), (#), ($), (%), (^), (&), ('), (.), (()), ({}), (_), ( ).

4)Also for {attr_key}/{attr_val}: 
attr_key		
It's not allowed to specify one attribute key twice per one request.
attr_key and attr_val
The header's length and count are not limited. There is a limit where Key + Value pairs in sum could not exceed 3MB length.
Attribute keys can contain letters (A–Z, a-z), numbers (0-9) and special characters.
attr_val		
It's allowed to specify one attribute key twice per one request.

**Comment by KirillovDenis** [comment]
> Also, a word `section` in headers: `wallet section`, `peers section`, `logger section` etc in [gate-configuration.md](https://github.com/nspcc-dev/neofs-http-gw/blob/master/docs/gate-configuration.md) looks redundant, doesn't it?

By `section` I meant section in yaml config (not section in `.md` file). In such interpretation it doesn't look redundant as for me.

**Comment by KirillovDenis** [comment]
> 3)And it would be great if you mentioned the limitations of NNS in something like this:

I'm not sure we have to mention such limitation because we don't create such name using http-gw. So http-gw doesn't check provided name

**Comment by alexvanin** [comment]
- Let's call it `api.md` as a first step to unify doc structure across the repos
- We should update it after we close #214

**Review by alexvanin — APPROVED** [review]

File: docs/api.md [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+**Note:** `cid` parameter can be base58 encoded container ID or container name
+(the name must be registered in NNS, see appropriate section in [README](../README.md#nns)).
+
+Route parameters can be:
+
+* `Single` - match a single path segment (cannot contain `/` and be empty)
+* `Catch-All` - match everything (such parameter usually the last one in routes)
+* `Query` - regular query parameter
+
+### Bearer token
```
[the remark this answers is not in the document]
Comment (KirillovDenis) [reply docs/api.md:19]: Kind of. I thought `##` too big for this particular header

**Review by alexvanin — APPROVED** [review]

**Pull request merged**